### ISSUE TITLE
Integrity check warnings when updating daily up to 9.1.1

### ISSUE THREAD

**Issue by PVince81** [owncloud/updater#379]
Using the test update server from https://github.com/owncloud/administration-internal/pull/32.

1. Start with 8.2.8
1. Set config.php to point to the test server
1. Switch channel to daily
1. Update: it will deliver 9.0.5
1. Update again: it will deliver 9.1.1 / daily stable9.1

```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- FILE_MISSING
		- lib/composer/bin/doctrine-dbal
		- updater/src/Resources/views/partials/error.php
	- INVALID_HASH
		- updater/app/bootstrap.php
		- updater/src/Console/Application.php
		- updater/src/Controller/IndexController.php

Raw output
==========
Array
(
    [core] => Array
        (
            [FILE_MISSING] => Array
                (
                    [lib/composer/bin/doctrine-dbal] => Array
                        (
                            [expected] => 
                            [current] => 
                        )

                    [updater/src/Resources/views/partials/error.php] => Array
                        (
                            [expected] => fbb1e810c5e4a22e87e604bfdec9a39b5ae1fe4daa04e9a2a26af774a2e004a723608f88cd25cf856da28b588c1445b766f489a9ea0258cdee6447872caf46b6
                            [current] => 
                        )

                )

            [INVALID_HASH] => Array
                (
                    [updater/app/bootstrap.php] => Array
                        (
                            [expected] => 80476362835c3941744669f417dfb3832319f4930d606526aa303fac583557b983bc665d7a856b23b7b7cf350b85b0b9cce8f136bbf0e2f3857ed03239ea8bad
                            [current] => d3e0bd2806af767a513e8776f0ab07f6c160ac430487752380b6f5c9c128039e113dde6eeabed4e0b6cb0341fe855e57f754e3eda0f6eac66744d4e03f7367c0
                        )

                    [updater/src/Console/Application.php] => Array
                        (
                            [expected] => e673de6a27cde432a08bfd1e68ffadfc5cd0025acbe2a1fc6158078caa01be744dcbefad0bf756b07591f96b499dd2bdcbec62cd1874fd2e463756723f112a5a
                            [current] => 55311f48f937f00e6fc7b37ecf90ff7ea5b127b1e16cd3d932be5c9d9222e2acbb55f4cbacf34de02396ad5a3494b981a51ed62a2c44ede2ac7472bfbf680dd6
                        )

                    [updater/src/Controller/IndexController.php] => Array
                        (
                            [expected] => 26e864e9684a5e8833f0c4194b20583fd36642c9b5fe316ed5574c19b8735cfd3e9c41adae326b62845a8a34a792fb0f2925bff1e09f8f43b53ebb8be9c0439e
                            [current] => 15072440f84163a03638edc8bc26211365398bd326c53986e1606494bbf38661be74fc7a9a2912787d34bf11d1977b4ef70b6724f918b16efacf4d9ac9eba14f
                        )

                )

        )

)
```

@VicDeo @davitol

**Comment by PVince81** [comment]
Now step by step:

- after updating the first time from 8.2.8 to 9.0.5 with daily channel:
```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- provisioning_api
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.

Raw output
==========
Array
(
    [provisioning_api] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )
)
```

I thought the error would be about 3rdparty, not sure why provisioning API now.

**Comment by PVince81** [comment]
- after updating the second time from 9.0.5 to 9.1.1 with daily channel, following above:
```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- updater/app/config/container.php
		- updater/src/Command/CheckSystemCommand.php
		- updater/src/Command/CheckpointCommand.php
		- updater/src/Command/ExecuteCoreUpgradeScriptsCommand.php
		- updater/src/Console/Application.php
		- updater/src/Utils/AppManager.php
		- updater/src/Utils/Checkpoint.php
		- updater/src/Utils/Fetcher.php
		- updater/src/Utils/Locator.php
		- updater/src/Utils/ZipExtractor.php
	- FILE_MISSING
		- updater/src/Tests/Utils/DocLinkTest.php
		- updater/src/Utils/DocLink.php
	- EXTRA_FILE
		- updater/src/Utils/ConfigWriter.php
		- updater/src/Utils/BzipExtractor.php

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [updater/app/config/container.php] => Array
                        (
                            [expected] => f3178bf2eea881bb92a93c617a12d02db667b31cc1abc164d4cce299fceda30bb34b7aa4414a89a574b7c75ac93e24532bb7b978477631df572b856e8bee1c94
                            [current] => a125a0085252cdde6706eee326087816ee5aea959500ddfe7a229357e6a2bbcaea92690b624a30c5ea7cfbaf8843ad93e4b7c95d9fc0d1dfc1bfab1d05bf6877
                        )

                    [updater/src/Command/CheckSystemCommand.php] => Array
                        (
                            [expected] => cc11aed423b76753b5aa85145e3138aff42018a2071b138f4a5477c6db669e1d54921c18a1a71582d4a038d04d08a0e7a59413b0af6397b3538e16949340c774
                            [current] => 208667946900423b604745ef172019eeb66b561b1707ce3ba08cf87a6e1f154a04372240f7aa1a76a22b4a18cec7dc00bb8cf06f3bb40f1b2968f646dd36b7c4
                        )

                    [updater/src/Command/CheckpointCommand.php] => Array
                        (
                            [expected] => ac249296ff6a93551dc79437f48fff3e1901416fb9ec1f08e4b3d9b7e855453ee36401f38d3f14477bece650a9d5f88b95fcb1bdbc6e9d42ec1bb11d90153c08
                            [current] => 748986d6bae6b8331691b8943b4899c2cefa66df6617547c597345ea0a1ff0cb7e85159ab1089ba0ce5a149f946c65d0225be7d8ad5984ef3ac4cec33bbbb33d
                        )

                    [updater/src/Command/ExecuteCoreUpgradeScriptsCommand.php] => Array
                        (
                            [expected] => fe4d73d495e2e6f223e56b261262f09a0cc58152681cae42815a489c696e7a7adfc2d3fc8440222c97587f867d2b69128b9194cdb6306aeaaf3c80ddc40c155d
                            [current] => 50e179bd31f9146a8dc88bcc44c696a882f26369e6c9f3f17e1b172b8304e1387519cc3f2b516926ca7118c8f9b4a3b1444f0c72a0746e3c8b695ee569ae7027
                        )

                    [updater/src/Console/Application.php] => Array
                        (
                            [expected] => 55311f48f937f00e6fc7b37ecf90ff7ea5b127b1e16cd3d932be5c9d9222e2acbb55f4cbacf34de02396ad5a3494b981a51ed62a2c44ede2ac7472bfbf680dd6
                            [current] => 7e56e2109920830f03879cc09f2c51602aac6c000c2a014061dc7aa806b8cfe687424b3bfcfb772b0f6f13a645eaaa8826acca26f75e5bbe51ef7c021cb8df9a
                        )

                    [updater/src/Utils/AppManager.php] => Array
                        (
                            [expected] => a77151de023e11e851889f13380e840008873b9326b0efd2807057deeb01a029b068b82adfeb686c722606bac6edeb081f3c24f1405c5def914e389d42e68e38
                            [current] => 5b7743330e762b61e5c3142be01c516623213c5ea6d648e417b0a850fbc4adab07a5f347d44a365b08adbe314953375e8e45b70de5034e1d77f93093eabb5184
                        )

                    [updater/src/Utils/Checkpoint.php] => Array
                        (
                            [expected] => 584ec051716f476a0d69fd2eb1e5af3bdb79ed3e6c7d312a74e11005e30b62804c2035563bd0b23027f2d2c4dde48393c109c9f1e1207babb590f027b4fe79c4
                            [current] => 4b2e42d1ec60b890ab822ef200283c7e04016d7dfe4fd99ace33dc0ddb9ebb9fb73deb98b0443050e7252eed24f522d38a85f2d8942dfc508f4ab536f3d44d5d
                        )

                    [updater/src/Utils/Fetcher.php] => Array
                        (
                            [expected] => 402fc10a28c2a5e3a1e8210db0ec51149b69cbfb9113050c2318e3507129dc3e1377dec2129d7e722269fb1874197c64f3a68c64f85e7382fc9960322c74b3c4
                            [current] => ffac8ed267723a42d6c106eb94cecd5cbc6bff0d76a4738c3e597a48cfcc79857609247a4e8d13ff091afd51c224349b52645829830c4e674cdcbbb06f573982
                        )

                    [updater/src/Utils/Locator.php] => Array
                        (
                            [expected] => 3309881f2aeaec70d5c21dd454f16470bc3ea316cdbd58e1cc4c9c89d2723612695de001f88434ad229088ac3c15ac9273b719471f33010d23290b15bb4d2a8c
                            [current] => 2f68144b3f2559fe901e2beb3338c2187174f3d66a96ba6b7ff2529126c42d7307b59398895cb1395cf7346a79b8390929128e58c7ade85644cddb0ad60651f0
                        )

                    [updater/src/Utils/ZipExtractor.php] => Array
                        (
                            [expected] => cc9845687f0987dab9c2c11983621e458cdf6d5d108d0286a2b41ba3fa04f51d11897680122ec235ebf1a54b6fe92ee256d774b9b09e1075b61479237d86698e
                            [current] => a9677633cb722dc00053b38d634a8fce4f65a7fff712199b2950a94648dd472eea63c09ec26a4ac0aa204b73d71fb6a6f5680e9086f63f518e20969b5672e5fc
                        )

                )

            [FILE_MISSING] => Array
                (
                    [updater/src/Tests/Utils/DocLinkTest.php] => Array
                        (
                            [expected] => 849ceded5f7f4e79d73d6212b490503e796b2c2b9ceb7944339fe289cd5b2fdee99e9169ce3574d00551293633af4512f3e31f4272bddb8e4f54844ede55b608
                            [current] => 
                        )

                    [updater/src/Utils/DocLink.php] => Array
                        (
                            [expected] => 8133139a302cfed577a51d491a56769c23dc45f2abafca0bd265540679b64275412525872eff22b197192d2c2ca085cf26403d20ac8b778cd279281b0e66e32e
                            [current] => 
                        )

                )

            [EXTRA_FILE] => Array
                (
                    [updater/src/Utils/ConfigWriter.php] => Array
                        (
                            [expected] => 
                            [current] => cbcb82fa7f809734b8c803810e653eb70326af5670be607785e63a670a3f8f592c8e4c277b76389e6d102c97974221e3875c59592a6325d0e9fbdf4bab213fbb
                        )

                    [updater/src/Utils/BzipExtractor.php] => Array
                        (
                            [expected] => 
                            [current] => 68956ff70b36c6a71eaae31e2d883a3e7523af23cccc1135826f6fdf31aa75e436d469e5e879421b9175163fd0eff67dac31327ebf6dcfc952132544fe4d45d9
                        )

                )

        )

)
```

Now let's check whether the beta channel has the same issue...

@VicDeo your help needed here

**Comment by PVince81** [comment]
Weird, on the beta channel it works fine, even though it's supposed to use the same tarballs as with the daily steps in-between.

Now I also tried beta update 8.2.8->9.0.5->9.1.1 which went fine and then I switched to daily, and now getting:
```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- FILE_MISSING
		- lib/composer/bin/doctrine-dbal
		- updater/src/Resources/views/partials/error.php
	- INVALID_HASH
		- updater/app/bootstrap.php
		- updater/src/Console/Application.php
		- updater/src/Controller/IndexController.php

Raw output
==========
Array
(
    [core] => Array
        (
            [FILE_MISSING] => Array
                (
                    [lib/composer/bin/doctrine-dbal] => Array
                        (
                            [expected] => 
                            [current] => 
                        )

                    [updater/src/Resources/views/partials/error.php] => Array
                        (
                            [expected] => fbb1e810c5e4a22e87e604bfdec9a39b5ae1fe4daa04e9a2a26af774a2e004a723608f88cd25cf856da28b588c1445b766f489a9ea0258cdee6447872caf46b6
                            [current] => 
                        )

                )

            [INVALID_HASH] => Array
                (
                    [updater/app/bootstrap.php] => Array
                        (
                            [expected] => 80476362835c3941744669f417dfb3832319f4930d606526aa303fac583557b983bc665d7a856b23b7b7cf350b85b0b9cce8f136bbf0e2f3857ed03239ea8bad
                            [current] => d3e0bd2806af767a513e8776f0ab07f6c160ac430487752380b6f5c9c128039e113dde6eeabed4e0b6cb0341fe855e57f754e3eda0f6eac66744d4e03f7367c0
                        )

                    [updater/src/Console/Application.php] => Array
                        (
                            [expected] => e673de6a27cde432a08bfd1e68ffadfc5cd0025acbe2a1fc6158078caa01be744dcbefad0bf756b07591f96b499dd2bdcbec62cd1874fd2e463756723f112a5a
                            [current] => 55311f48f937f00e6fc7b37ecf90ff7ea5b127b1e16cd3d932be5c9d9222e2acbb55f4cbacf34de02396ad5a3494b981a51ed62a2c44ede2ac7472bfbf680dd6
                        )

                    [updater/src/Controller/IndexController.php] => Array
                        (
                            [expected] => 26e864e9684a5e8833f0c4194b20583fd36642c9b5fe316ed5574c19b8735cfd3e9c41adae326b62845a8a34a792fb0f2925bff1e09f8f43b53ebb8be9c0439e
                            [current] => 15072440f84163a03638edc8bc26211365398bd326c53986e1606494bbf38661be74fc7a9a2912787d34bf11d1977b4ef70b6724f918b16efacf4d9ac9eba14f
                        )

                )

        )

)
```

the version is ownCloud 9.2.0 pre alpha (daily) Build:2016-09-22T03:01:06+00:00 e357c2c7fe4bb345b37fa53f677297c9c9b147f8 so maybe there is a problem on master

**Comment by VicDeo** [comment]
@davitol integrity failure for 9.0.[0,1] -> 9.0.4 is known and fixing itself with 9.0.4 -> 9.0.5
https://github.com/owncloud/core/issues/25544#issuecomment-246779516

**Comment by PVince81** [comment]
@VicDeo yes indeed for stable.
What about the daily ones related to 9.1.1 ?

**Comment by VicDeo** [comment]
@PVince81 
For 8.2.8->9.0.5: see https://github.com/owncloud/updater/pull/384
9.0.5->9.1.1: Can't reproduce, according to the the set of extra/missing files the whole `updater` dir is still at 9.0.x

**Comment by VicDeo** [comment]
@PVince81 something wrong with a signature itself for daily master:
```
# grep false core/signature.json 
        "lib\/composer\/bin\/doctrine-dbal": false,
        "lib\/composer\/bin\/generate_vcards": false,
        "lib\/composer\/bin\/naturalselection": false,
        "lib\/composer\/bin\/sabredav": false,
        "lib\/composer\/bin\/vobject": false,
```

as you see, there is `false` instead of hash. I guess the list of files for signature is collected before `lib/composer/bin` is removed and the script tries to sign these files  after the removal

**Comment by PVince81** [comment]
I retried my original steps and when updating to 9.0.5 (as part of the daily channel in the updater update pending deployment), now seeing these:
```
Results
=======
- provisioning_api
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.

Raw output
==========
Array
(
    [provisioning_api] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )
)
```

Rescanning doesn't help.

What is so different with the daily channel that makes integrity checks like these fail ?

**Comment by PVince81** [comment]
the diff shows that the provisioning app wasn't update, here is the first file from the diff:
``'
diff -r owncloudtest-9.0.5/apps/provisioning_api/appinfo/info.xml owncloudtest/apps/provisioning_api/appinfo/info.xml
15a16
>       <shipped>true</shipped>
20c21
<       <version>0.4.1</version>
---
>       <version>0.3.0</version>
22c23,24
<               <prevent_group_restriction/>
---
>               
>               <filesystem/>
25c27
<               <owncloud min-version="9.0" max-version="9.0" />
---
>               <owncloud min-version="8.2" />
```

**Comment by PVince81** [comment]
Uh oh... trouble ahead...

I've now updated from 8.2.8 to 9.0.5 using the beta channel (on the pending update server). The update goes fine and it doesn't show any integrity errors.
HOWEVER a diff of the code with the tarball shows the same problem as above: the provisioning API app has not been updated ! And from what I see the file "signature.json" is missing since that app's code is still from 8.2.8.

Strangely though, when using the daily channel the signature file is there in "apps/provisioning_api/appifno" but other files were not updated.

@VicDeo

**Comment by PVince81** [comment]
Raised https://github.com/owncloud/core/issues/26463 to improve integrity check to properly detect such cases.

**Comment by PVince81** [comment]
So this means that for some reason the updater from 8.2.8 skipped the provisioning api. Is there an app whitelist that needs updating ?

**Comment by PVince81** [comment]
Will retest with https://github.com/owncloud/updater/pull/405, it is likely that the integrity failures will be gone.

- [ ] TODO: raise bug about 8.2.8 updater not updating provisioning API

**Comment by VicDeo** [comment]
8.2.8 updater https://github.com/owncloud/updater/issues/410

@PVince81 Does patching according to https://central.owncloud.org/t/patches-for-known-update-issues-from-version-9-0-4-9-0-5-9-1-0-9-1-1/3549 help with the other issues from this thread?

**Comment by PVince81** [comment]
When reaching 9.1.1 from 9.0.5 I received an integrity check warning that was solved by a rescan, so it must be the known issue that is solved in 9.0.6. Then updating to daily worked fine.

Closing.
